core/src/main/java/hudson/model/Actionable.java

```diff
@@ -95,7 +102,15 @@
         for (TransientActionFactory<?> taf : ExtensionList.lookup(TransientActionFactory.class)) {
             if (taf.type().isInstance(this)) {
                 try {
-                    _actions.addAll(createFor(taf));
+                    Collection<? extends Action> newActions = createFor(taf);
+                    for (Action action : newActions) {
+                        if (action instanceof Action) {
+                            _actions.add(action);
+                        } else {
+                            LOGGER.log(Level.WARNING, "Actionable#getAllActions: Found a not action: " + action.getClass().toString());
```

Comment:
1. Appending `"message " + someObject.toString()` is pointless; you could just use `"message " + someObject` instead.
1. You really meant to use `Class.getName`. `toString` prints a `"class "` prefix which is unwanted here.